osu.Game/Screens/Play/GameplayState.cs

```diff
@@ -46,9 +46,12 @@ public class GameplayState
         public bool HasPassed { get; set; }
 
         /// <summary>
-        /// Whether the user failed during gameplay. This is only set when the gameplay session has completed due to the fail.
+        /// Whether the user failed during gameplay and the fail animation has been displayed.
         /// </summary>
-        public bool HasFailed { get; set; }
+        /// <remarks>
+        /// In multiplayer, this is never set to <c>true</c> even if the player reached zero health, due to <see cref="PlayerConfiguration.AllowFailAnimation"/> being turned off.
+        /// </remarks>
+        public bool ShownFailAnimation { get; set; }
```

Comment:
I disagree so much with this change and everything it implies.

*Any* gameplay logic being conditional on "being shown the fail animation" is completely and utterly wrong.

This feels like going in a completely off-base direction. And I think the fix to this entire issue should be _local to the multiplayer flow_.

Reply:
I approached this initially with the thought that `Player` has two separate states of failing, each being:
 1. marking score as failed / F rank (determined by `HealthProcessor.HasFailed`)
 2. presenting fail animation and stopping gameplay (determined by `GameplayState.HasFailed`, never set to true in multiplayer since https://github.com/ppy/osu/pull/26384).

Therefore I've went with renaming `GameplayState.HasFailed` to the name in the diff of this conversation, as well as using that state for whether the judgement processors should no longer apply new results.

I will be honest that I didn't pay enough attention at the diff and missed the fact that `GameplayState.HasFailed` is used in more places than usual, and having the name specifically referring to "presence of fail animation" is indeed sketchy.

I could use better names such as `HasStoppedDueToFail`, but I've went ahead with your alternative proposal in https://github.com/ppy/osu/pull/26751/commits/64b61108ad5b389ce136385393eb3eba9a4f9bc8 (making the solution completely local) if discussion for this is not necessary right now (I could imagine another player screen that would require this sort of fix, namely `ReplayPlayer` when replaying failing multiplayer scores).

Reply:
> I could imagine another player screen that would require this sort of fix, namely `ReplayPlayer` when replaying failing multiplayer scores

I think this is out of scope for now because we currently don't even have a way to tell whether a replay is from multiplayer or from solo. So this has been an issue for a *long* time now, and it is one caused by the fact that the *health processor* is allowed to fail a replay, while not considering the fact that in some circumstances the conditions for failure may not be the same.